stdlib/http/server.pyi

```diff
@@ -50,20 +52,33 @@ class BaseHTTPRequestHandler(socketserver.StreamRequestHandler):
     def parse_request(self) -> bool: ...  # Undocumented
 
 class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
+    server_version: str
     extensions_map: Dict[str, str]
     if sys.version_info >= (3, 7):
         def __init__(
             self,
             request: bytes,
             client_address: Tuple[str, int],
             server: socketserver.BaseServer,
-            directory: Optional[StrPath] = ...,
+            directory: StrPath | None = ...,
         ) -> None: ...
     else:
         def __init__(self, request: bytes, client_address: Tuple[str, int], server: socketserver.BaseServer) -> None: ...
     def do_GET(self) -> None: ...
     def do_HEAD(self) -> None: ...
+    def send_head(self) -> io.BytesIO | BinaryIO | None: ...  # undocumented
+    def list_directory(self, path: StrPath) -> io.BytesIO | None: ...  # undocumented
+    def translate_path(self, path: StrPath) -> StrPath: ...  # undocumented
```

Comment:
`path.split()` is called and the return value in constructed using `os.path.join()`:

```suggestion
    def translate_path(self, path: str) -> str: ...  # undocumented
```